enterprise/internal/database/perms_store.go

```diff
@@ -281,6 +314,34 @@ func (s *permsStore) LoadUserPermissions(ctx context.Context, p *authz.UserPermi
 	return nil
 }
 
+func (s *permsStore) FetchReposByUserAndExternalAccount(ctx context.Context, userID int32, accountID int32) (ids []api.RepoID, err error) {
+	const format = `
+SELECT repo_id
+FROM user_repo_permissions
+WHERE user_id = %s
+  AND user_external_account_id = %s;
+`
+
+	q := sqlf.Sprintf(format, userID, accountID)
+
+	ctx, save := s.observe(ctx, "FetchReposByUserAndExternalService", "")
+	defer func() {
+		save(&err)
+	}()
+
+	repos, err := basestore.ScanInt32s(s.Query(ctx, q))
+	if err != nil {
+		return nil, errors.Wrap(err, "scanning repo ids")
+	}
+
+	ids = make([]api.RepoID, 0, len(repos))
+	for _, id := range repos {
+		ids = append(ids, api.RepoID(id))
+	}
+
+	return ids, nil
+}
+
```

Comment:
```suggestion
func (s *permsStore) FetchReposByExternalAccount(ctx context.Context, accountID int32) (ids []api.RepoID, err error) {
	const format = `
SELECT repo_id
FROM user_repo_permissions
WHERE user_external_account_id = %s;
`

	q := sqlf.Sprintf(format, accountID)

	ctx, save := s.observe(ctx, "FetchReposByExternalAccount", "")
	defer func() {
		save(&err)
	}()

	repos, err := basestore.ScanInt32s(s.Query(ctx, q))
	if err != nil {
		return nil, errors.Wrap(err, "scanning repo ids")
	}

	ids = make([]api.RepoID, 0, len(repos))
	for _, id := range repos {
		ids = append(ids, api.RepoID(id))
	}

	return ids, nil
}

```

We don't need a user ID, since an external account should map to a single user ID.